Plugin/Rest/Api.php

```diff
@@ -150,4 +163,42 @@ protected function isAuthorizationRequest($path)
     {
         return preg_match('/integration\/(admin|customer)\/token/', $path) !== 0;
     }
+
+    /**
+     * @param array<mixed> $requestHeader
+     * @return string
+     */
+    protected function getIntegrationName($requestHeader)
+    {
+        if(!$this->config->isIntegrationNameEnabled()){
+            return '';
+        }
+        if(!isset($requestHeader['data'])){
+            return '';
+        }
+        try{
+            $data = explode(',', $requestHeader['data']);
+            $data = array_map(function($keyValue){
+                list($key, $value) = explode('=', $keyValue);
+                return [
+                    'key' => $key,
+                    'value' => trim($value, '"')
+                ];
+            }, $data);
+            $data = array_combine(
+                array_column($data, 'key'),
+                array_column($data, 'value')
+            );
+            $consumerKey = $data['oauth_consumer_key'];
```

Comment:
I would prefer an approach that uses existing methods for parsing the data (and in the end using less code). Do you think something in this direction would produce the same effect? 

```suggestion
            $consumerKey = array_reduce(
	        explode(',', $requestHeader['data']),
	        function($key, $keyValue) {
		    parse_str($keyValue, $data);
		    return $key ?: (array_key_exists('oauth_consumer_key', $data) ? trim(reset($data), '"') : null);
	        });
```